kinsumer.go

```diff
@@ -112,6 +112,7 @@ func NewWithInterfaces(kinesis kinesisiface.KinesisAPI, dynamodb dynamodbiface.D
 		clientID:              uuid.NewV4().String(),
 		clientName:            clientName,
 		config:                config,
+		leaderLost:            make(chan bool),
```

Comment:
Best practice for signalling channels is to use `chan struct{}`

Reply:
That's a micro optimization that relies on the gc compiler's behaviour of not allocating for empty structs (the spec makes no guarantees <sup>1</sup>). and `chan bool` was already used for `stoprequest` so I wanted to be consistent. Anyway, I've changed it.

<details><sup>1</sup> <a href="https://golang.org/ref/spec#Size_and_alignment_guarantees">Two distinct zero-size variables may have the same address in memory.</a></details>

Reply:
Thanks for the informative link, I did not know that. :)